addons/marketing_card/tests/test_campaign.py

```diff
@@ -323,3 +323,56 @@ def test_campaign_ownership(self):
         with self.assertRaises(exceptions.AccessError):
             campaign_as_other.unlink()
         campaign_as_owner.unlink()
+
+    def test_mail_render_security(self):
+        """Asserts the security of the QWeb rendering of the field body_html of card.campaign
+
+        The security is based on the fact:
```

Comment:
Would simply link _check_access_right_dynamic_template docstring to explain once and completely :)